internal/component/discovery/target.go

```diff
@@ -188,7 +188,17 @@ func (t Target) ConvertInto(dst interface{}) error {
 		})
 		*dst = result
 		return nil
+	case *map[string]string:
```

Comment:
This was the code path required to fix the issue with 
```
ts=2025-02-25T12:04:02.660000145Z level=error msg="failed to evaluate config" controller_path=/ controller_id="" node=prometheus.exporter.snmp.snmp err="decoding configuration: /etc/alloy/config.alloy:204:16: discovery.file.snmp.targets[0] target::ConvertInto: conversion to '*map[string]string' is not supported"
```
see test for more details